services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/component/BoosterNotificationStructureProvider.java

```diff
@@ -0,0 +1,77 @@
+import app.coronawarn.server.services.distribution.assembly.structure.WritableOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.archive.ArchiveOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.archive.decorator.signing.DistributionArchiveSigningDecorator;
+import app.coronawarn.server.services.distribution.assembly.structure.file.FileOnDisk;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig;
+import app.coronawarn.server.services.distribution.dgc.BusinessRule.RuleType;
+import app.coronawarn.server.services.distribution.dgc.DigitalGreenCertificateToCborMapping;
+import app.coronawarn.server.services.distribution.dgc.exception.DigitalCovidCertificateException;
+import app.coronawarn.server.services.distribution.dgc.exception.FetchBusinessRulesException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ * Reads Booster notification rules from the respective files in the class path or from dcc client and build a {@link
+ * BoosterNotificationStructureProvider} with them.
+ */
+
+@Component
+public class BoosterNotificationStructureProvider {
+
+  private static final Logger logger = LoggerFactory.getLogger(BoosterNotificationStructureProvider.class);
+
+  public static final String EXPORT_BINARY_FILENAME = "export.bin";
+
+  private final DistributionServiceConfig distributionServiceConfig;
+  private final CryptoProvider cryptoProvider;
+  private final DigitalGreenCertificateToCborMapping dgcToCborMapping;
+
+  /**
+   * Create an instance.
+   */
+  public BoosterNotificationStructureProvider(DistributionServiceConfig distributionServiceConfig,
+      CryptoProvider cryptoProvider, DigitalGreenCertificateToCborMapping dgcToCborMapping) {
+    this.distributionServiceConfig = distributionServiceConfig;
+    this.cryptoProvider = cryptoProvider;
+    this.dgcToCborMapping = dgcToCborMapping;
+  }
+
+  /**
+   * Returns the publishable archive with Booster Notification Business rules Cbor encoded structures.
+   */
+  public Writable<WritableOnDisk> getBoosterNotificationRules() {
+    return getBnRulesArchive(RuleType.BoosterNotification,
+        distributionServiceConfig.getDigitalGreenCertificate().getBoosterNotification());
+  }
+
+  /**
+   * Create business rules Archive. If any exception is thrown during fetching data and packaging process, an empty
+   * Archive will be published in order to not override any previous archive on CDN with broken data. Provided rules are
+   * filtered by rule type parameter which could be 'Acceptance', 'Invalidation' or 'BoosterNotification'.
+   *
+   * @param ruleType    - rule type to receive rules for
+   * @param archiveName - archive name for packaging rules
+   * @return - business rules archive
+   */
+  private Writable<WritableOnDisk> getBnRulesArchive(RuleType ruleType, String archiveName) {
```

Comment:
I would go with full BoosterNotification name in here also in order to keep consistency with all the namings in this class. Also for clarity.